### PR TITLE
[Impeller] move more aiks tests to DL.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Part of https://github.com/flutter/flutter/issues/142054

### CODE REVIEW

File: impeller/display_list/aiks_dl_runtime_effect_unittests.cc [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+  builder.ClipRRect(
+      SkRRect::MakeRectXY(SkRect::MakeXYWH(0, 0, 400, 400), 10.0, 10.0),
+      DlCanvas::ClipOp::kIntersect);
+  builder.DrawRect(SkRect::MakeXYWH(0, 0, 400, 400), paint);
+  builder.Restore();
+
+  ASSERT_TRUE(OpenPlaygroundHere(builder.Build()));
+}
+
+TEST_P(AiksTest, DrawPaintTransformsBounds) {
```
Comment (jonahwilliams) [inline impeller/display_list/aiks_dl_runtime_effect_unittests.cc:65]: Something went wrong with this one.

File: impeller/display_list/aiks_dl_runtime_effect_unittests.cc [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    AiksTest* test,
+    std::string_view name,
+    const std::shared_ptr<std::vector<uint8_t>>& uniform_data = {},
+    const std::vector<std::shared_ptr<DlColorSource>>& samplers = {}) {
+  auto runtime_stages =
+      test->OpenAssetAsRuntimeStage("runtime_stage_example.frag.iplr");
+
+  auto runtime_stage = runtime_stages[PlaygroundBackendToRuntimeStageBackend(
+      test->GetBackend())];
+
```
Comment (flar) [inline impeller/display_list/aiks_dl_runtime_effect_unittests.cc:32]: Is it worth copying over the asserts from the original code?

File: impeller/display_list/aiks_dl_runtime_effect_unittests.cc (hunk 1 above)
Comment (flar) [reply impeller/display_list/aiks_dl_runtime_effect_unittests.cc:65]: I can't spot any differences... :(

**Review by flar — APPROVED** [review]
LGTM once you find the source of what went wrong with the second test...

File: impeller/display_list/aiks_dl_runtime_effect_unittests.cc (hunk 1 above)
Comment (flar) [reply impeller/display_list/aiks_dl_runtime_effect_unittests.cc:65]: I see the file "gradient.frag.iplr" in 2 different places in the `out/.../gen` directory.

```
../out/host_debug_unopt_arm64/gen/flutter/impeller/fixtures/gradient.frag.iplr
../out/host_debug_unopt_arm64/gen/flutter/impeller/fixtures/assets/gradient.frag.iplr
```

Comment (flar) [reply impeller/display_list/aiks_dl_runtime_effect_unittests.cc:65]: But I only see one `gradient.frag` in the git tree so they should have both been generated from the same source...

Comment (flar) [reply impeller/display_list/aiks_dl_runtime_effect_unittests.cc:65]: And their contents are identical...

File: impeller/display_list/aiks_dl_runtime_effect_unittests.cc (hunk 2 above)
Comment (jonahwilliams) [reply impeller/display_list/aiks_dl_runtime_effect_unittests.cc:33]: Done

File: impeller/display_list/aiks_dl_runtime_effect_unittests.cc [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+using namespace flutter;
+
+namespace {
+std::shared_ptr<DlRuntimeEffectColorSource> MakeRuntimeEffect(
+    AiksTest* test,
+    std::string_view name,
+    const std::shared_ptr<std::vector<uint8_t>>& uniform_data = {},
+    const std::vector<std::shared_ptr<DlColorSource>>& samplers = {}) {
+  auto runtime_stages =
+      test->OpenAssetAsRuntimeStage("runtime_stage_example.frag.iplr");
```
Comment (jonahwilliams) [inline impeller/display_list/aiks_dl_runtime_effect_unittests.cc:28]: oops

File: impeller/display_list/aiks_dl_runtime_effect_unittests.cc (hunk 1 above)
Comment (jonahwilliams) [reply impeller/display_list/aiks_dl_runtime_effect_unittests.cc:65]: Fixed...

File: impeller/display_list/aiks_dl_runtime_effect_unittests.cc (hunk 3 above)
Comment (flar) [reply impeller/display_list/aiks_dl_runtime_effect_unittests.cc]: D'oh!

**Pull request merged**